trajectory_executor/com_serial.py

```diff
@@ -0,0 +1,222 @@
+import rclpy
+from rclpy.node import Node
+
+from trajectory_interfaces.srv import SensorDataRequest
+from trajectory_interfaces.msg import JointAngles
+
+import serial
+import struct
+import json
+
+from trajectory_executor.datastructures import serial_read_states, data_packet_struct
+
+
+class SerialConnection(Node):
+  """Manages a serial connection between ROS and Arduino"""
+  def __init__(self):
+    super().__init__('serial_connection')
+
+    self.preamble_length = 4
+    self.data_byte_length = 2
+
+    self._service = self.create_service(
+      SensorDataRequest,
+      'sensor_data_request',
+      self.sensor_data_request_callback
+    )
+
+    self._subscriber = self.create_subscription(
+      JointAngles,
+      'joint_angles',
+      self.send_joint_angles_callback,
+      10
+    )
+
+    self.arduino_port = serial.Serial(
+      port="/dev/ttyAMA1",  # TODO: evaluate if we should change to cli flag
+      baudrate=115200
+    )
+
+  def sensor_data_request_callback(self, request: SensorDataRequest, response: SensorDataRequest):
+    """Ping Arduino for sensor data"""
+    request_sensors = 1
+    checksum = 255 - (request_sensors * self.data_byte_length) % 256
+
+    # send request for sensor data
+    for i in range(self.preamble_length):
+      self.arduino_port.write(struct.pack('>B', 255))
```

Comment:
noticed that `'>B'` is used a lot. is it a standard? either way feels like a magic string, maybe we should have a comment or make it into a named variable

Reply:
I learned about this when working on my previous robot. The `'>B'` tells `struct.pack` how to encode the data (in this case encode into bytes). There might be a better/cleaner way to encode the data as bytes, but this is the first solution I found a while back for my other robot and I just stuck with it. We could make it a named variable for organization.

Reply:
Add this as a comment so it's understandable. For the future, consider adding comments to things that might not be so apparent.

Reply:
Updated with added comment